internal/bindings/waf_dl.go

```diff
@@ -43,33 +44,25 @@ type wafSymbols struct {
 // The caller is responsible for calling wafDl.Close on the returned object once they
 // are done with it so that associated resources can be released.
 func NewWafDl() (dl *WafDl, err error) {
-	var file string
-	file, err = lib.DumpEmbeddedWAF()
+	var handle uintptr
+	file, err := lib.DumpEmbeddedWAF()
 	if err != nil {
 		return
 	}
 	defer func() {
-		rmErr := os.Remove(file)
-		if rmErr != nil {
-			if err == nil {
-				err = rmErr
-			} else {
-				// TODO: rely on errors.Join() once go1.20 is our min supported Go version
-				err = fmt.Errorf("%w; along with an error while removing %s: %v", err, file, rmErr)
-			}
+		if rmErr := os.Remove(file); rmErr != nil {
+			err = errors.Join(err, fmt.Errorf("error removing %s: %w", file, rmErr))
 		}
 	}()
 
-	var handle uintptr
```

Comment:
Why move this declaration higher up, when it's actually only useful starting there?

Reply:
Done